core/gui/src/app/workspace/component/file-selection/file-selection.component.ts

```diff
@@ -14,9 +14,11 @@ import { parseFilePathToDatasetFile } from "../../../common/type/dataset-file";
   styleUrls: ["file-selection.component.scss"],
 })
 export class FileSelectionComponent implements OnInit {
-  readonly datasets: ReadonlyArray<DashboardDataset> = inject(NZ_MODAL_DATA).datasets;
```

Comment:
try to keep using readonly array for data fetched from backend. although don't inject it with some mock data.

Reply:
changed